lib/dal/src/attribute/value.rs

```diff
@@ -2397,30 +2419,66 @@ impl AttributeValue {
+                // create a node with these out of sync, or remove the ordering node from
+                // object props altogether.
+                PropKind::Object => {
+                    // NOTE probably can get the unordered ones if it comes down to it.
+                    let child_ids = Self::get_child_av_ids_from_ordering_node(ctx, id)
+                        .await?
+                        .ok_or(AttributeValueError::NoOrderingNodeForAttributeValue(id))?;
+                    let child_prop_ids = Prop::direct_child_prop_ids_ordered(ctx, prop.id).await?;
+
+                    // Get the mapping from PropId -> AttributeValueId
+                    let mut av_prop_map = HashMap::with_capacity(child_ids.len());
+                    for &child_id in &child_ids {
+                        let child_prop_id = Self::prop_id_for_id_or_error(ctx, child_id).await?;
+                        av_prop_map.insert(child_prop_id, child_id);
+                    }
 
-        if let Some(ordering) = workspace_snapshot
-            .outgoing_targets_for_edge_weight_kind(id, EdgeWeightKindDiscriminants::Ordering)
-            .await?
-            .pop()
-        {
-            let node_weight = workspace_snapshot.get_node_weight(ordering).await?;
-            if let NodeWeight::Ordering(ordering_weight) = node_weight {
-                Ok(ordering_weight
-                    .order()
-                    .clone()
-                    .into_iter()
-                    .map(|ulid| ulid.into())
-                    .collect())
-            } else {
-                Err(AttributeValueError::NodeWeightMismatch(
-                    ordering,
-                    NodeWeightDiscriminants::Ordering,
-                ))
+                    // For each PropId (in schema order), look up the AttributeValueId
+                    let child_ids_in_prop_order: Vec<AttributeValueId> = child_prop_ids
+                        .iter()
+                        .filter_map(|child_prop_id| av_prop_map.get(child_prop_id).copied())
+                        .collect();
+
+                    // Make sure we actually returned all the children
+                    if child_ids_in_prop_order.len() != child_ids.len() {
+                        for child_id in child_ids {
+                            if !child_ids_in_prop_order.contains(&child_id) {
+                                return Err(AttributeValueError::FieldNotChildOfObject(child_id));
+                            }
+                        }
+                        // Unreachable: child_ids_in_prop_order can only be <= av_prop_map.
+                        return Err(AttributeValueError::Unreachable);
+                    }
+
+                    Ok(child_ids_in_prop_order)
+                }
             }
         } else {
-            // Leaves don't have ordering nodes
-            Ok(vec![])
+            // TODO can this ever happen? Seems like AttributeValueError::MissingPropEdge()
```

Comment:
Probably not

Reply:
I can update this to just return the error--it would make things simpler for sure.

Reply:
(I'll make a quick followup PR.)